test-suite-java-spring/src/main/java/io/micronaut/openapi/spring/api/MyController.java

```diff
@@ -0,0 +1,83 @@
+package io.micronaut.openapi.spring.api;
+
+// tag::imports[]
+import io.micronaut.http.annotation.RequestBean;
+import io.swagger.v3.oas.annotations.Parameter;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import static io.swagger.v3.oas.annotations.enums.ParameterIn.QUERY;
+
+// end::imports[]
+// tag::clazz[]
+@RestController
+@RequestMapping("/my-api")
+public class MyController {
+
+    @PostMapping(value = "/testMultipart/{pathVar}", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public void testMultipart(@RequestBean WrappedRq rq) {
```

Comment:
not sure that forcing spring users to use a Micronaut specific feature makes sense? Why is this being recommended in the docs?

Reply:
This is necessary so that micronaut openapi understands that there is a wrapper class parameter in front of it, and inside there can be both parts of a multipart request, and path variables, and variables from the request.

Spring does not have a special annotation to parse everything correctly - it does this without any additional annotations, unlike micronaut. But openapi cannot understand what kind of parameter is in front of it, so it needs to be marked somehow for correct processing. In order not to create an additional annotation, I suggest using the existing RequestBean annotation.

If you look at springdoc - they encountered the same problem and were forced to add their own annotation to mark such parameters - `@ParameterObject`